### DISCUSSION TITLE
Complying with Pylint C0121 produces strange results

### DISCUSSION THREAD

**Discussion by tribumirkov** [sqlalchemy/sqlalchemy#8378] [Ideas]
Hey guys,

I have SQLAlchemy==1.4.39 installed and Pylint suggests I should have:

foos = session.execute(
    select(Foo).
    where(
        Foo.bar==bar,
        **Foo.disabled is False**,
    )
).scalars().first()

instead of

foos = session.execute(
    select(Foo).
    where(
        Foo.bar==bar,
        **Foo.disabled==False**,
    )
).scalars().first()

but the former does not produce the expected result.

Why is this so? Will this change in 2.0?

Thank you!
Niko

**Comment by CaselIT** [comment]
Hi,

> Why is this so? 

Usually when comparing a boolean it's best to use `is` instead of `==`, but the operation performed by `is` cannot be overridden so there is no way of generating a boolean expression like when doing `==`

> Will this change in 2.0?

No, not unless python introduces the ability of overriding the `is` operator

**Reply by tribumirkov** [reply]
Grazie mille, Federico! 

Could you just spend a few words on what "cannot be overridden" means in this case?

Thank you again!

**Reply by CaselIT** [reply]
You can find more information here https://docs.python.org/3/reference/datamodel.html

Basically most of the type when you do something in python what actually happens is that a method on an object is called. For example `==` will call a method called `__eq__`. Sqlalchemy overrides the `__eq__` method to return a boolean expression that will be compiled to an sql statement.
The is operator has no equivalent `__is__` method, so sqlalchemy cannot change the default behaviour of python, so the default check will be run and instead of a boolean operation that is compiled to an sql statement you get a boolean False (that is compied to False, but it's not what you want)

**Reply by tribumirkov** [reply]
super interesting, thank you again!

**Comment by zzzeek** [comment]
for this purpose we have an [is_()](https://docs.sqlalchemy.org/en/14/core/sqlelement.html#sqlalchemy.sql.expression.ColumnOperators.is_) operator, which you can use in places that the linter wants you to use the `is` operator.

**Reply by tribumirkov** [reply]
great stuff, I already implemented it and it works like a charm! thank you, Mike!
